django_airavata/apps/api/views.py

```diff
@@ -861,7 +697,46 @@ def get_instance(self, lookup_value):
             self.authz_token, lookup_value)
 
     def perform_create(self, serializer):
-        return self.request.airavata_client.createGroupResourceProfile(self, self.authz_token, serializer.save())
+        group_resource_profile = serializer.save()
+        group_resource_profile.gatewayId = self.gateway_id
+        group_resource_profile_id = self.request.airavata_client.createGroupResourceProfile(authzToken=self.authz_token,
+                                                                                            groupResourceProfile=group_resource_profile)
+        group_resource_profile = self.request.airavata_client.getGroupResourceProfile(
+            self.authz_token, group_resource_profile_id)
+        serializer.instance = group_resource_profile
+
+    def perform_update(self, serializer):
+        self.request.airavata_client.updateGroupResourceProfile(self.authz_token, serializer.save())
+
+
+class SharedEntityGroups(APIBackedViewSet):
+    serializer_class = serializers.SharedGroups
 
     def perform_update(self, serializer):
-        return self.request.airavata_client.updateGroupResourceProfile(self, self.authz_token, serializer.save())
\ No newline at end of file
+        before = serializer.data
+        after = serializer.initial_data
+        del serializer._data
+        remaining = [item for item in before["groupList"] if item not in after["groupList"]]
+        params = dict(after)
+        params["domainId"] = settings.GATEWAY_ID
+        params["permissionTypeId"] = READ_PERMISSION_TYPE.format(settings.GATEWAY_ID)
+        params["cascadePermission"] = True
+        ret = self.request.sharing_client.shareEntityWithGroups(**params)
+        if remaining:
+            self.request.sharing_client.revokeEntitySharingFromGroups(domainId=settings.GATEWAY_ID,
+                                                                      entityId=before["entityId"], groupList=remaining,
+                                                                      permissionTypeId=READ_PERMISSION_TYPE.format(
+                                                                          settings.GATEWAY_ID))
+        serializer.save()
+
+    def get_instance(self, entity_id):
+        groups = {
+            'entityId': entity_id
+        }
+        group_list = self.request.sharing_client.getListOfSharedGroups(domainId=settings.GATEWAY_ID, entityId=entity_id,
```

Comment:
We need to add a method like getAllAccessibleUsers to Airavata API. We shouldn't call the sharing service directly. But that's okay for now. I'll make an issue.